Lecture4/Task4/Voucher.java

```diff
@@ -0,0 +1,65 @@
+import java.util.Locale;
+
+import predefinedExceptions.RegexException;
+
+abstract class Voucher {
+
+    private String name;
+    private double price;
+    private String country;
+    private String startDate;
+    private int countDays;
+    private int discount;
+    private double cost;
+    private String formattedCost;
+
+    Voucher(String[] voucherDetails) throws RegexException {
+        this.name = voucherDetails[1];
+        this.price = Double.parseDouble(voucherDetails[2]);
+        if (voucherDetails[3].matches("[A-Z][A-Z]|[A-Z][A-Z][A-Z]")) {
+            this.country = voucherDetails[3];
+        } else {
+            throw new RegexException();
```

Comment:
bad tone to throw an exception without details. You didn't provide any error message

Reply:
fixed in Tasks1-4-fixed package
Error message will be provided when exception will be catched and processed